src/http/routes.cc

```diff
@@ -37,6 +37,17 @@ void verify_param(const request& req, const sstring& param) {
     }
 }
 routes::routes() : _general_handler([this](std::exception_ptr eptr) mutable {
+    try {
+        std::rethrow_exception(eptr);
+    } catch (const redirect_exception& _e) {
+        auto rep = std::make_unique<reply>();
+        rep->add_header("Location", _e.url).set_status(_e.status()).done(
+                "json");
+        return rep;
+    } catch (...) {
+        return exception_reply(std::current_exception());
```

Comment:
should this fall through and return the exception ptr?

Reply:
Yeah, that is going to give a slightly neater backtrace.  Updated.